codegen/method.go

```diff
@@ -186,9 +192,16 @@ func NewMethod(
 	method.setValidStatusCodes()
 
 	if method.HTTPMethod == "GET" && method.RequestType != "" {
-		return nil, errors.Errorf(
-			"invalid annotation: HTTP GET method with body type",
-		)
+		if !method.IsEndpoint {
+			return nil, errors.Errorf(
+				"invalid annotation: HTTP GET method cannot have a body",
+			)
+		}
+
+		err := method.setQueryParamStatements(funcSpec)
```

Comment:
Nit: better change the signature of `setQueryParamStatements` to 

```
func queryParamStatements(*compile.FunctionSpec) ([]string, error)
```

since the func doesn't have to depend on `MethodSpec` type.